servers/rendering/shader_language.cpp

```diff
@@ -4086,6 +4086,286 @@ Variant ShaderLanguage::constant_value_to_variant(const Vector<ShaderLanguage::C
 	return Variant();
 }
 
+Variant ShaderLanguage::get_default_datatype_value(DataType p_type, int p_array_size, ShaderLanguage::ShaderNode::Uniform::Hint p_hint) {
+	int array_size = p_array_size;
+
+	Variant value;
+	switch (p_type) {
+		case ShaderLanguage::TYPE_BOOL:
+			if (array_size > 0) {
+				PackedInt32Array array;
+				for (int i = 0; i < array_size; i++) {
+					array.push_back(false);
+				}
+				value = Variant(array);
+			} else {
+				value = Variant(false);
+			}
+			break;
```

Comment:
Don't we already have code for this in core / Variant to generate default initialized variants of each type? Seems like it could simplify the code a lot to reuse that (see `VariantDefaultInitializer` etc.).

Reply:
It doesn't work, it still fill it with a null value. Strange.

Reply:
Looks like VariantDefaultInitializer just change a raw memory but does not change a type of Variant.